sql/events.go

```diff
@@ -0,0 +1,239 @@
+// Copyright 2023 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sql
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const EventTimeStampFormat = "2006-01-02 15:04:05"
+
+// EventDetails are the details of the event.
+type EventDetails struct {
+	Name                 string
+	Definer              string
+	OnCompletionPreserve bool
+	Status               EventStatus
+	Comment              string
+	Definition           string
+
+	// events' ON SCHEDULE clause fields
+	ExecuteAt    time.Time
+	HasExecuteAt bool
+	ExecuteEvery *EventOnScheduleEveryInterval
+	Starts       time.Time
+	HasStarts    bool
+	Ends         time.Time
+	HasEnds      bool
+
+	// time values of event create/alter/execute metadata
+	Created        time.Time
+	LastAltered    time.Time
+	LastExecuted   time.Time
+	ExecutionCount uint64
+	// TODO: add TimeZone
+}
+
+// GenerateCreateEventStatement returns CREATE EVENT statement constructed from EventDetails.
+// It's important to construct new CREATE EVENT statement instead of using the initial statement
```

Comment:
Just use the first sentence as the comment for this method, this is too much detail and doesn't help a client understand what the method does